### PR TITLE
dbeaver: 21.0.0 -> 21.0.1

### PR DESCRIPTION
**Pull request by jojosch**
###### Motivation for this change
Update to latest release (https://github.com/dbeaver/dbeaver/releases/tag/21.0.1). 

Also replaced the manual cp'ing of the desktop File with `copyDesktopItems` and fixed some `nixpkgs-hammering` warnings.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
```
/nix/store/mxpkpr80rzlfsbwpcjxzgp6f31ljj6pp-dbeaver-ce-21.0.0	 1355389656
/nix/store/pvf120fpv8y025qcvn0ix25c4mj4ndpv-dbeaver-ce-21.0.1	 1355492848
```
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/dbeaver/default.nix [hunk 1]
```diff
@@ -62,6 +53,12 @@ stdenv.mkDerivation rec {
     outputHash = "sha256-xKlFFQXd2U513KZKQa7ttSFNX2gxVr9hNsvyaoN/rEE=";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/dbeaver/default.nix:53]: That probably requires an update, too.

Comment (jojosch) [reply pkgs/applications/misc/dbeaver/default.nix:53]: hmm, i did set both sha256's to an invalid checksum but this one does not change. Will investigate that tonight.

Comment (jojosch) [reply pkgs/applications/misc/dbeaver/default.nix:53]: I ran the `buildPhase` inside two clean VMs (one with 21.0.0 and one with 21.0.1 checked out). 

The only difference in the resulting .m2 directories are the `_remote.repositories` files which are cleaned by the `installPhase` and contain the current timestamp/datetime.

**Comment by lourkeur** [comment]
Result of `nixpkgs-review pr 117145` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dbeaver</li>
  </ul>
</details>

output works for me

File: pkgs/applications/misc/dbeaver/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/dbeaver/default.nix:53]: Then they did not change deps.

**Pull request merged**